src/sql/workbench/services/notebook/common/interface.ts

```diff
@@ -0,0 +1,16 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import * as azdata from 'azdata';
+
+export interface INotebookInput {
+	defaultKernel: azdata.nb.IKernelSpec,
+	connectionProfile: azdata.IConnectionProfile,
+	isDirty(): boolean;
+	setDirty(boolean);
+}
+
+export function isINotebookInput(value: any): value is INotebookInput {
+	return true;
```

Comment:
You should actually be checking that it's a notebook input here...that would mean verifying that it has all the properties of INotebookInput and that they're of the correct type.